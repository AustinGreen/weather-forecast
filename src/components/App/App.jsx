import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TransparentBackground from '../../shared/Backgrounds';
import HomeHeading from '../../shared/Headings';
import {
  PATH_BASE,
  PARAM_TYPE,
  PARAM_UNIT,
  PARAM_APPID,
  PATH_FORECAST,
  PARAM_DAYS,
} from '../../constants';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      isError: false,
      forecastResult: '',
    };

    // Bind class methods to object instances
    this.onSearchSubmit = this.onSearchSubmit.bind(this);
    this.onSearchChange = this.onSearchChange.bind(this);
    this.onDataRecieved = this.onDataRecieved.bind(this);
  }

  onSearchChange(e) {
    this.setState({ searchTerm: e.target.value });
  }

  onDataRecieved(forecastResult) {
    const { searchTerm } = this.state;
    const { history } = this.props;
    if (forecastResult.cod === '404') {
      this.setState({ isError: true });
    } else {
      this.setState({ forecastResult });
      history.push(`/forecast/${searchTerm}`, forecastResult);
    }
  }

  onSearchSubmit(e) {
    const { searchTerm } = this.state;
    if (searchTerm.length > 0) {
      fetch(
        `${PATH_BASE}${PATH_FORECAST}${searchTerm}&${PARAM_TYPE}&${PARAM_UNIT}&${PARAM_APPID}&${PARAM_DAYS}`,
      )
        .then(response => response.json())
        .then(forecastResult => this.onDataRecieved(forecastResult));
    }
    e.preventDefault();
  }

  render() {
    return (
      <TransparentBackground
        className="section column is-half
          is-offset-one-quarter is-large has-text-centered"
      >
        <HomeHeading className="title">Enter your city and state</HomeHeading>
        <form onSubmit={this.onSearchSubmit} className="field has-addons has-addons-centered">
          <p className="control">
            <input
              onChange={this.onSearchChange}
              className={this.state.isError ? 'input is-danger' : 'input'}
              type="text"
              placeholder="New York, NY"
            />
          </p>
          <p className="control">
            <button type="submit" className="button is-primary">
              Search
            </button>
          </p>
        </form>
      </TransparentBackground>
    );
  }
}

App.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default App;
