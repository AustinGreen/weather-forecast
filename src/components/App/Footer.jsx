import React from 'react';
import FontAwesome from 'react-fontawesome';
import styled from 'styled-components';

const StyledFooterContainer = styled.footer`
  background-color: #AABE3D;
`;

const Footer = () => (
  <StyledFooterContainer className="footer is-hidden-mobile is-hidden-tablet-only">
    <div className="container">
      <div className="content has-text-centered">
        <p>
          <span className="icon is-large">
            <a href="http://github.com/AustinGreen" target="_blank" rel="noopener noreferrer">
              <FontAwesome name="github" size="3x" style={{ color: '#010101' }} />
            </a>
          </span>
        </p>
      </div>
    </div>
  </StyledFooterContainer>
);

export default Footer;
