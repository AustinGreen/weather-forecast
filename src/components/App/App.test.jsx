/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
