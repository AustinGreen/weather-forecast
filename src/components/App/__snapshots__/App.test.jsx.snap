exports[`App snapshots 1`] = `
<div>
  <div
    className="bGOOgi">
    <div
      className="djzNwS">
      <div
        className="idhBkf" />
      <div
        className="jppquj">
        <div
          className="lblJGx" />
      </div>
      <div
        className="XcSzE">
        <div
          className="lblJGx" />
      </div>
      <div
        className="bVomTb">
        <div
          className="lblJGx" />
      </div>
      <div
        className="LdiPp">
        <div
          className="lblJGx" />
      </div>
      <div
        className="kaGqkq">
        <div
          className="lblJGx" />
      </div>
      <div
        className="giRQOq">
        <div
          className="lblJGx" />
      </div>
    </div>
    <div
      className="section column is-half is-offset-one-quarter is-large has-text-centered gQTaUs">
      <h1
        className="title iFFOKQ">
        Enter your city and state
      </h1>
      <form
        className="field has-addons has-addons-centered"
        onSubmit={[Function]}>
        <p
          className="control">
          <input
            className="input"
            onChange={[Function]}
            placeholder="New York, NY"
            type="text" />
        </p>
        <p
          className="control">
          <button
            className="button is-primary"
            type="submit">
            Search
          </button>
        </p>
      </form>
    </div>
  </div>
  <footer
    className="footer is-hidden-mobile is-hidden-tablet-only jPLOcw">
    <div
      className="container">
      <div
        className="content has-text-centered">
        <p>
          <span
            className="icon is-large">
            <a
              href="http://github.com/AustinGreen"
              rel="noopener noreferrer"
              target="_blank">
              <span
                aria-hidden={true}
                className="fa fa-github fa-3x"
                style={
                  Object {
                    "color": "#010101",
                  }
                } />
            </a>
          </span>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
